apps/dashboard/app/apps/nav_bar.rb

```diff
@@ -0,0 +1,134 @@
+class NavBar
+
+  def self.items(nav_config)
+    nav_config.map do |nav_item|
+      if nav_item.is_a?(String)
+        item_from_token(nav_item)
+      elsif nav_item.is_a?(Hash)
+        if nav_item.fetch(:links, nil)
+          extend_item(nav_menu(nav_item), 'layouts/nav/group')
+        elsif nav_item.fetch(:url, nil)
+          extend_item(nav_link(nav_item, nil, nil), 'layouts/nav/link')
+        elsif nav_item.fetch(:apps, nil)
+          matched_apps = nav_apps(nav_item, nil, nil)
+          extend_item(matched_apps.first.links.first, 'layouts/nav/link') if matched_apps.first && matched_apps.first.links.first
+        elsif nav_item.fetch(:template, nil)
+          nav_template(nav_item)
```

Comment:
I'm not sure if we want this. This is, you could render _any_ template? Do you have a use case for this?

Reply:
The use case for this was to add existing links into the navigation like:
- Interactive Sessions
- All Apps
- Featured Apps

I wasn't sure how to call it and implement it, but template was the easy solution.

The other idea is the one you mentioned, having like a `link_code` that we can use to do a lookup. Like:

`link_code: all_apps`
`all_apps => render template: /layouts/nav/all_apps`

Reply:
Got it, yea referencing templates directly could be dangerous. I think the lookup solution could work, but I'm not too sure about `link_code` and whether that makes sense to the reader...

Maybe it could just fit into the string block?
```yaml
nav_bar:
  - all_apps
```